docs/nav2_overview.md

```diff
@@ -0,0 +1,146 @@
+autonomy. Nav2 comes with two navigators: `NavigateToPose` and `NavigateThroughPoses`. Both navigators load behavior
+trees from XML files, and users can customize each navigator's functionality through these XML files. Additionally,
+users can provide their own navigators if they desire.
+
+For an example, consider the following behavior tree specification:
+
+```xml
+<root BTCPP_format="4" main_tree_to_execute="MainTree">
+  <BehaviorTree ID="MainTree">
+    <PipelineSequence name="NavigateWithReplanning">
+      <ControllerSelector selected_controller="{selected_controller}" default_controller="FollowPath" topic_name="controller_selector"/>
+      <PlannerSelector selected_planner="{selected_planner}" default_planner="GridBased" topic_name="planner_selector"/>
+      <DistanceController distance="1.0">
+        <ComputePathToPose goal="{goal}" path="{path}" planner_id="{selected_planner}" error_code_id="{compute_path_error_code}"/>
+      </DistanceController>
+      <FollowPath path="{path}" controller_id="{selected_controller}" error_code_id="{follow_path_error_code}"/>
+    </PipelineSequence>
+  </BehaviorTree>
+</root>
+```
+
+From this specification, the `NavigateToPose` navigator will behave as follows:
+
+1. Make a service call to the Controller Server to activate the specified controller plugin. (See `params.yaml` for
+   more information.)
+2. Make a service call to the Planner Server to activate to specified planner plugin. (See `params.yaml` for more
+   information.)
+3. Make an action call to the Planner Server to generate a new reference path to track. The navigator will request a
+   new reference path whenever the robot is greater than 1 meter from the goal pose.
+4. Make an action call to the Controller Server to track the reference path.
+
+The navigator will repeat the pipeline until the robot reaches its gaol pose.
+
+> [!NOTE]
+> The Planning and Control Servers' plugins do the actual planning and tracking, respectively. The navigator, which
+> lives within the BT Navigator Server, only decides when to send the corresponding ROS 2 action requests.
+
+## Planning Server
+
+This server contains a collection of path planner plugins that generate a path from the robot's current pose to some
+goal pose. The plugins expose a ROS 2 action interface for generating a path through free-space. Users can choose
+one of the planners that comes with Nav2, or they can bring their own.
+
+The server maintains an understanding of the robot's environment through a global costmap.
+
+## Controller Server
+
+This server contains a collection of controller plugins that command the robot to follow the provided reference path.
+Each plugin exposes a ROS 2 action. Similar to the Planning Server, users can choose from several provided controller
+plugins, or they can implement their own.
+
+They are responsible for publishing `geometry_msgs::msg::Twist` messages onto the `/cmd_vel` topic.
+
```

Comment:
If it is important to mention that the Planning Server maintains an understanding of the robot's environment through a global costmap, maybe it's worth mentioning that the Controller Server maintains this through a local costmap? Might not be needed since this is already mentioned earlier in the document.